atlas/src/shared/components/VideoPlayer/VideoPlayer.tsx

```diff
@@ -5,31 +5,90 @@ import { useVideoJsPlayer, VideoJsConfig } from './videoJsPlayer'
 type VideoPlayerProps = {
   className?: string
   autoplay?: boolean
+  isInBackground?: boolean
+  playing?: boolean
+  onDataLoaded?: () => void
 } & VideoJsConfig
 
-const VideoPlayer: React.FC<VideoPlayerProps> = ({ className, autoplay, ...videoJsConfig }) => {
+const VideoPlayer: React.FC<VideoPlayerProps> = ({
+  className,
+  autoplay,
+  isInBackground,
+  playing,
+  onDataLoaded,
+  ...videoJsConfig
+}) => {
   const [player, playerRef] = useVideoJsPlayer(videoJsConfig)
   const [playOverlayVisible, setPlayOverlayVisible] = useState(true)
+  const [initialized, setInitialized] = useState(false)
+
+  const displayPlayOverlay = playOverlayVisible && !isInBackground
 
   useEffect(() => {
-    if (!player || !autoplay) {
+    if (!player) {
       return
     }
 
-    const handler = async () => {
-      try {
-        await player.play()
-      } catch (e) {
-        console.warn('Autoplay failed:', e)
-      }
+    const handler = () => {
+      setInitialized(true)
     }
 
     player.on('loadstart', handler)
 
     return () => {
       player.off('loadstart', handler)
     }
-  }, [player, autoplay])
+  }, [player])
+
+  useEffect(() => {
+    if (!player) {
+      return
+    }
+
+    const handler = () => {
+      if (onDataLoaded) {
+        onDataLoaded()
+      }
+    }
+
+    player.on('loadeddata', handler)
+
+    return () => {
+      player.off('loadeddata', handler)
+    }
+  }, [player, onDataLoaded])
+
+  useEffect(() => {
+    if (!player || !initialized || !autoplay) {
+      return
+    }
+
+    const playPromise = player.play()
+    if (playPromise) {
+      playPromise.catch((e) => {
+        console.warn('Autoplay failed:', e)
+      })
+    }
+  }, [player, initialized, autoplay])
+
+  useEffect(() => {
+    if (!player) {
+      return
+    }
+
+    if (playing != null) {
+      if (playing) {
```

Comment:
I think this second if is not necessary given the one above

Reply:
First `if` is meant only to check if the value is `undefined` or `null`, the second one will check for `true` / `false`

Reply:
I see, lets leave it as is then.